### ISSUE TITLE
sap*scaleout: fix incorrect regex for hostname handling

### ISSUE THREAD

**Issue by sean-freeman** [sap-linuxlab/ansible.playbooks_for_sap#35]
Fix hardcoded hostnames search to regex
  - Removed hardcoded reject for all hostnames containing '0'
  - Replaced with regex search checking for "ending with 0" instead.

```yaml
# Before
reject('search', '0')

# After
reject('regex', '0$')
```

**Comment by RishabhKodes** [comment]
Fix for this, https://github.com/sap-linuxlab/ansible.playbooks_for_sap/pull/36

**Comment by sean-freeman** [comment]
Resolved in PR #39
